tests/test_circuit.py

```diff
@@ -1,146 +1,171 @@
+        observables_funcs_dict = {
+            'X' : qml.PauliX, 
+            'Y' : qml.PauliY,
+            'Z' : qml.PauliZ,
+            'H' : qml.Hadamard
+        }
+        cls.observables = {
+            args.qubit_index[i] : observables_funcs_dict[args.observables[i]]
+            for i in range(len(args.qubit_index))
+        }
+        cls.ansatze = [circ.Sim14, circ.Sim15, circ.StronglyEntangling]
+        circuits_exp_value = [
+            ansatz(
+                cls.n_qubits, cls.n_layers,
+                cls.axis_embedding, cls.observables, False
+            ) for ansatz in cls.ansatze
+        ]
+        circuits_probs = [
             ansatz(
                 cls.n_qubits, cls.n_layers,
-                cls.axis_encoding, cls.observables) for ansatz in cls.ansatze
+                cls.axis_embedding, cls.observables, True
+            ) for ansatz in cls.ansatze
         ]
+        cls.circuits = [circuits_exp_value, circuits_probs]
         cls.params = [
             torch.nn.Parameter(
                 torch.randn(
-                    circuit.parameters_shape)) for circuit in cls.circuits
+                    circuit.parameters_shape)) for circuit in cls.circuits[0]
         ]
         cls.input = torch.nn.Parameter(torch.randn(cls.n_qubits))
+        cls.results_circuits = [[], []]
+        for i,c in enumerate(cls.circuits):
+            for j,circuit in enumerate(c):
+                    circuit_function = circuit.build_circuit_function
+                    cls.results_circuits[i].append(
+                        circuit.run_and_measure_circuit(
+                        circuit_function)(cls.input, cls.params[j])
+                    )
 
-
-    def test_circuits_output_correct_type(self):
+    def test_circuits_output_correct_type(self)-> None:
         """
-        Test the that the the three different ansatze implemented 
-        work. To do so it will be asserted that the expected value obtained
-        after running the circuit is a tensor
+        Test the that the three different ansatze implemented 
+        work (for expectation value and prob outputs),
+        asserting that a tensor is obtained as output.
         """
-        for i,circuit in enumerate(self.circuits):
-            circuit_function = circuit.build_circuit_function
-            results = circuit.run_and_measure_circuit(
-                circuit_function)(self.input, self.params[i])
-            ### We make sure the output has the proper format
-            for r in results:
-                self.assertIsInstance(r, torch.Tensor)
+        for i,r in enumerate(self.results_circuits):
+            for result in r:
+                with self.subTest(result=result):
+                    for tensor in result:
+                        self.assertIsInstance(tensor, torch.Tensor)
 
-    def test_result_of_different_rescaling_functions(self):
+    def test_circuit_outputs_probabilities_between_0_and_1(self)-> None:
```

Comment:
This is a very good test!
It makes sense, has a good name and is well-documented.